### PR TITLE
Make mimikatz inside zip and extract only if config says so

### PR DESCRIPTION
**Pull request by itaymmguardicore**
# Feature / Fixes
> Mimikatz is now inside password protected zip, and extracted only if config says so.

* [X] Have you added an explanation of what your changes do and why you'd like to include them?
* [X] Have you successfully tested your changes locally?

### CODE REVIEW

File: infection_monkey/monkey.spec [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
+import zipfile
+
+# Name of zip file that will be created
+MIMIKATZ_ZIP_NAME = 'tmpzipfile123456.zip'
+# Name of zip file in monkey. That's the name of the file in the _MEI folder
+MIMIKATZ_ZIP_NAME_MONKEY = MIMIKATZ_ZIP_NAME
+# Name of mimikatz dll in zip archive
+MIMIKATZ_DLL_NAME_ZIP = 'tmpzipfile123456.dll'
+# Password for mimikatz zip
+MIMIKATZ_ZIP_PASSWORD = 'HEDFGFDSgfsdg4235342#@$^@#shd35'
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply infection_monkey/monkey.spec]: This is my mistake. This shouldn't have been committed.

File: infection_monkey/readme.txt [hunk 2]
```diff
@@ -70,4 +70,8 @@ Sambacry requires two standalone binaries to execute remotely.
 
 Mimikatz is required for the Monkey to be able to steal credentials on Windows. It's possible to either compile from sources (requires Visual Studio 2013 and up) or download the binaries from 
 https://github.com/guardicore/mimikatz/releases/tag/1.0.0
-Download both 32 and 64 bit DLLs and place them under [code location]\infection_monkey\bin
\ No newline at end of file
+Download both 32 and 64 bit zipped DLLs and place them under [code location]\infection_monkey\bin
+Alternatively, if you build Mimikatz, put each version in a zip file.
+1. The zip should contain only the Mimikatz DLL named tmpzipfile123456.dll
+2. It should be protected using the password 'VTQpsJPXgZuXhX6x3V84G'.
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply infection_monkey/readme.txt]: Spec file is wrong. This one is right

File: infection_monkey/system_info/mimikatz_collector.py [hunk 3]
```diff
@@ -13,12 +16,30 @@ class MimikatzCollector(object):
...
     """
 
+    # Name of Mimikatz DLL. Must be name of file in Mimikatz zip.
+    MIMIKATZ_DLL_NAME = 'tmpzipfile123456.dll'
+
+    # Name of ZIP containing Mimikatz. Must be identical to one on monkey.spec
+    MIMIKATZ_ZIP_NAME = 'tmpzipfile123456.zip'
+
+    # Password to Mimikatz zip file
+    MIMIKATZ_ZIP_PASSWORD = r'VTQpsJPXgZuXhX6x3V84G'
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply infection_monkey/system_info/mimikatz_collector.py]: Spec file won't have the password any more.
The readme is the only other place the password is gonna be in (and we can't import it there 🙃)

File: infection_monkey/system_info/mimikatz_collector.py [hunk 4]
```diff
@@ -13,12 +16,30 @@ class MimikatzCollector(object):
...
+
     def __init__(self):
+        self._config = __import__('config').WormConfiguration
+        self._isInit = False
+        self._dll = None
+        self._collect = None
+        self._get = None
+        self.init_mimikatz()
+
+    def init_mimikatz(self):
```
[the remark this answers is not in the document]
Comment (itaymmguardicore) [reply infection_monkey/system_info/mimikatz_collector.py]: Nope. Don't see a reason there should be

File: infection_monkey/system_info/mimikatz_collector.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (danielguardicore) [reply infection_monkey/system_info/mimikatz_collector.py]: Shame :) 🥇

**Pull request merged**